com.zsmartsystems.zigbee/src/main/java/com/zsmartsystems/zigbee/ZigBeeNetworkManager.java

```diff
@@ -1011,15 +1018,19 @@ public void removeNetworkStateListener(ZigBeeNetworkStateListener stateListener)
     public synchronized void setTransportState(final ZigBeeTransportState state) {
         logger.debug("ZigBeeNetworkManager transport state updated to {}", state);
 
-        if (!validTransportStateTransitions.get(networkState).contains(state)) {
-            logger.debug(
-                    "Ignoring invalid transport state transition in ZigBeeNetworkState.{} by ZigBeeTransportState.{}",
-                    networkState, state);
-            return;
+        // Filter out unwanted transport state changes
+        synchronized (validTransportStateTransitions) {
```

Comment:
Since ```validTransportStateTransitions``` is final it can be used for synchronisation. It is not the map itself I'm worried about synchronising - it's the change to ```transportState``` - I can't synchronised on ```transportState``` as it is not a final field.

Reply:
Ah OK, thanks for the explanation, that certainly makes sense.